src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -92,7 +92,7 @@ class IqPolicyEvaluatorUtil
       throw e
     }
     else {
-      listener.logger.println Messages.IqPolicyEvaluation_UnableToCommunicate(e.message)
+      listener.logger.println Messages.IqPolicyEvaluation_Failed(e.message, e.cause.message)
```

Comment:
On what basis can we assume that the exception will always have a cause? 

If it's possible that it may not, we may need to change this both to avoid an NPE here, such as by use of the Groovy "Elvis operator" (I kid you not), as well as choosing the string template depending on whether or not we have a cause to display.

(I really don't know as I don't know anything about this code, just wanting to make sure we checked?)

Reply:
> On what basis can we assume that the exception will always have a cause?

it should always have a cause.  But maybe i should go on the safe side.

Reply:
> it should always have a cause. But maybe i should go on the safe side.

If you're sure it should always have a cause but not 100% sure, you could probably use the aforementioned Elvis operator. That should ensure that you don't get an NPE at least, and the message will just have "cause: null" in it if I'm not mistaken.

Reply:
> On what basis can we assume that the exception will always have a cause?

Changed 4f42179

Reply:
> Changed 4f42179

This also seems like a reasonable approach but I might just suggest having the one message and gracefully dealing with a `null`, rather than having two separate messages for a scenario that we don't expect would ever happen.

I might also suggest adding a test to check this behavior, either way, just as you already have for the current log message (with cause).